### PR TITLE
(#1084) Merged/Joined - allows to merge/join several java.util.Properties

### PR DESCRIPTION
**Pull request by Umbrah**
This is for #1084 
I added Merged extends MapEnvelope implementation and tests for it

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1122](https://github.com/yegor256/cactoos/issues/1122) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1122](https://github.com/yegor256/cactoos/issues/1122) is assigned to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), here is [why](https://www.0crat.com/footprint/C63314D6Z/37230af8-5cfd-4740-a87c-352837d79057); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/map/MergedTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ * Test case for {@link Merged}.
+ *
+ * @since 1.0
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class MergedTest {
+
+    @Test
+    public void behavesAsMap() {
+        new Assertion<>(
```
Comment (ilyakharlamov) [inline src/test/java/org/cactoos/map/MergedTest.java]: You forgot to call affirm() so the test does not test anything.

File: src/test/java/org/cactoos/map/MergedTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                    new MapEntry<>(1, 1)
+                )
+            ),
+            new BehavesAsMap<>(1, 1)
+        );
+    }
+
+    @Test
+    public void createsMapFromMaps() {
+        new Assertion<>(
```
Comment (ilyakharlamov) [inline src/test/java/org/cactoos/map/MergedTest.java]: You forgot to call affirm() so the test does not test anything.

File: src/test/java/org/cactoos/map/MergedTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                    new MapEntry<>(0, 0),
+                    new MapEntry<>(1, 1)
+                )
+            )
+        );
+    }
+
+    @Test
+    public void overridesValues() {
+        new Assertion<>(
```
Comment (ilyakharlamov) [inline src/test/java/org/cactoos/map/MergedTest.java]: You forgot to call affirm() so the test does not test anything.

**Comment by Umbrah** [comment]
@ilyakharlamov Fixed

**Comment by ilyakharlamov** [comment]
@Umbrah please add a couple of edge test cases that involve empty maps `new MapOf<>()` and, perhaps, null

**Comment by Umbrah** [comment]
@ilyakharlamov I added tests, please check. Small comment: I cannot use Assertion with Throws for null test, because Throws doesn't support null msg(which we have with NullPointerException), I will add an issue for it in cactoos-matchers project.

**Comment by ilyakharlamov** [comment]
@llorllale good to merge, FYI: https://github.com/llorllale/cactoos-matchers/issues/119

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 29 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job [#1122](https://github.com/yegor256/cactoos/issues/1122) is not in the agenda of @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), can't inspect

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1122](https://github.com/yegor256/cactoos/issues/1122) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0cat quality good

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)
